modules/server/Server.cpp

```diff
@@ -4,34 +4,137 @@
 
 #include <fstream>
 
-Connection *Server::simulate_connect(int socket) {
-	Connection *connection = new Connection(socket, sockaddr());
-	connections[socket] = connection;
-	return connection;
+#define MAX_BUFFER_LENGTH 4096
+
+Connection *Server::addConnection(const struct pollfd &connection) {
+	Connection *newConnect = new Connection(connection);
+	connections.push_back(newConnect);
+	return newConnect;
 }
 
-User *Server::simulate_join(Connection *connection, string name) {
-	User *u = new User(connection);
-	u->setNickName(name);
-	Irc::getInstance().addUser(u);
-	return u;
+Server::Server(const string& name, const string& port, const string& password) : name(name), port(port), password(password) {
```

Comment:
Change `port` from `const string&` to `int` type.
```suggestion
Server::Server(const string& name, int port, const string& password) : name(name), port(port), password(password) {
```